examples/machine-learning/a3-highgpu-8g/a3high-slurm-blueprint.yaml

```diff
@@ -0,0 +1,711 @@
+                regexp: '^GRUB_DEFAULT='
+                line: 'GRUB_DEFAULT="{{ tcpx_menu_entry.stdout }}"'
+              notify: Update GRUB
+            - name: Freeze tpcx kernel
+              ansible.builtin.dpkg_selections:
+                name: "{{ item }}"
+                selection: hold
+              loop:
+              - linux-image-gcp-tcpx
+              - linux-headers-gcp-tcpx
+            handlers:
+              - name: Update GRUB
+                ansible.builtin.command: update-grub
+
+      - type: ansible-local
+        destination: install_gcc_12.yml
+        content: |
+          ---
+          - name: Install GCC + GCC++ v12
+            hosts: all
+            become: true
+            tasks:
+            - name: Install GCC and G++ version 12
+              ansible.builtin.apt:
+                name:
+                - gcc-12
+                - g++-12
+                state: present
+                update_cache: yes
+
+            - name: Set GCC and G++ version 12 as default
+              community.general.alternatives:
+                name: gcc
+                path: /usr/bin/gcc-12
+                link: /usr/bin/gcc
+                priority: 120
+                slaves:
+                  - name: g++
+                    link: /usr/bin/g++
+                    path: /usr/bin/g++-12
+                  - name: gcov
+                    link: /usr/bin/gcov
+                    path: /usr/bin/gcov-12
+
+      - type: ansible-local
+        destination: configure_gpu_packages.yml
+        content: |
+          ---
+          - name: Install NVIDIA Packages and Configure Ops Agent
+            hosts: all
+            become: true
+            vars:
+              distribution: "{{ ansible_distribution | lower }}{{ ansible_distribution_version | replace('.','') }}"
+              package_url: https://developer.download.nvidia.com/compute/cuda/repos/{{ distribution }}/x86_64/cuda-keyring_1.1-1_all.deb
+              package_filename: /tmp/{{ package_url | basename }}
+              nvidia_packages:
+              - cuda-toolkit-12-8
+              - nvidia-dkms-570-server-open
+              - nvidia-driver-570-server-open
+              - datacenter-gpu-manager
+              - libnvidia-nscq-570
+              - nvidia-container-toolkit
+            tasks:
+            - name: Download NVIDIA repository package
+              ansible.builtin.get_url:
+                url: "{{ package_url }}"
+                dest: "{{ package_filename }}"
+            - name: Install NVIDIA repository package
+              ansible.builtin.apt:
+                deb: "{{ package_filename }}"
+                state: present
+            - name: Install NVIDIA Packages
+              ansible.builtin.apt:
+                name: "{{ item }}"
+                update_cache: true
+              loop: "{{ nvidia_packages }}"
+            - name: Freeze NVIDIA drivers and CUDA
+              ansible.builtin.dpkg_selections:
```

Comment:
here you are simply setting whatever version was installed on hold. I am not sure it is worse, if we are confident it will work, it has the advantage that it is the latest... But that also means the possibility of drift. WDYT?